File: 001-print-primes/go/findPrimes.go
package main

import "fmt"

func findPrimes(max int) []int {
	var primes []int
	for dividend := 1; dividend <= max; dividend++ {
		isPrime := true
		for dividor := 2; dividor < dividend; dividor++ {
			intResult := dividend / dividor
			converted := float32(intResult)
			floatResult := float32(dividend) / float32(dividor)
			if converted == floatResult {
				isPrime = false
				break
			}
		}
		if isPrime {
			primes = append(primes, dividend)
		}
	}
	return primes
}

func main() {
	primes := findPrimes(100)
	fmt.Println(primes)
}

File: 001-print-primes/2-ruby/findPrimes.rb
def findPrimes(max)
    primes = []
    for dividend in 1...max
        isPrime = true
        for dividor in 2...dividend
            if dividend / dividor == dividend.to_f / dividor.to_f
                isPrime = false
                break
            end
        end
        if isPrime
            primes.append(dividend)
        end
    end
    return primes
end

primes = findPrimes(100)
puts primes
